src/app/hearing/services/possible-hearing-actions.service.ts

```diff
@@ -0,0 +1,105 @@
+import { Injectable } from '@angular/core';
+import { MatDialog } from '@angular/material';
+import { HearingActions } from '../models/hearing-actions';
+import { DialogWithActionsComponent } from '../../features/notification/components/dialog-with-actions/dialog-with-actions.component';
+import { DEFAULT_DIALOG_CONFIG } from '../../features/transactions/models/default-dialog-confg';
+import { HearingService } from './hearing.service';
+import { TransactionDialogComponent } from '../../features/transactions/components/transaction-dialog/transaction-dialog.component';
+import { Hearing } from '../models/hearing';
+import { PossibleActionConfig } from '../models/possible-action-config';
+
+@Injectable()
+export class PossibleHearingActionsService {
+
+    possibleActions = {
+        [HearingActions.Unlist]: {
+            enabled: false,
+            openDialog: () => this.openUnlistDialog(),
+            callService: (hearing) => this.hearingService.unlist(hearing),
+            summaryText: 'Unlist hearing parts from session'
+        } as PossibleActionConfig,
```

Comment:
instead of casting each obj at the end. just add type to possibleActions property